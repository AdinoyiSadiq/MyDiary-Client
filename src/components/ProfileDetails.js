import React from 'react';

const ProfileDetails = (props) => {
  const { profile: { firstname, lastname, email, entryCount} } = props;
  return (
    <article className="profileSection">
      <div className="myProfile" id="profile">
        <header>
          <h1 id="fullname">{`${firstname} ${lastname}`}</h1>
        </header>
        <div className="firstName"><b>Name:</b> <span id="firstname">{`${firstname}`}</span></div>
        <div className="lastName"><b>Surname:</b> <span id="lastname">{`${lastname}`}</span></div>
        <div className="email"><b>Email:</b> <span id="email">{`${email}`}</span></div>
        <div className="entriesNumber"><b>Entries:</b> <span id="entryno">{`${entryCount}`}</span></div>
      </div>
    </article>
    );
};

export default ProfileDetails;