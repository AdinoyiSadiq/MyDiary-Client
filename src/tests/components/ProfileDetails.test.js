import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileDetails from '../../components/ProfileDetails';

let wrapped;

const profile = {
  firstname: 'firstname',
  lastname: 'lastname',
  email: 'email',
  entryCount: 'entry count'
}

beforeEach(() => {
  wrapped = shallow(<ProfileDetails profile={profile} />);
});


describe('ProfileDetails UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});