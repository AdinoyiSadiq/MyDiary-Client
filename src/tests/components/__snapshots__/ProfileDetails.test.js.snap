// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetails UI render features view should render as expected 1`] = `
<article
  className="profileSection"
>
  <div
    className="myProfile"
    id="profile"
  >
    <header>
      <h1
        id="fullname"
      >
        firstname lastname
      </h1>
    </header>
    <div
      className="firstName"
    >
      <b>
        Name:
      </b>
       
      <span
        id="firstname"
      >
        firstname
      </span>
    </div>
    <div
      className="lastName"
    >
      <b>
        Surname:
      </b>
       
      <span
        id="lastname"
      >
        lastname
      </span>
    </div>
    <div
      className="email"
    >
      <b>
        Email:
      </b>
       
      <span
        id="email"
      >
        email
      </span>
    </div>
    <div
      className="entriesNumber"
    >
      <b>
        Entries:
      </b>
       
      <span
        id="entryno"
      >
        entry count
      </span>
    </div>
  </div>
</article>
`;
